src/ui/connector-view/js/connector-view.js

```diff
@@ -0,0 +1,100 @@
+var flock = fluid.registerNamespace("flock");
+
+fluid.defaults("flock.midi.connectorView", {
+    gradeNames: ["flock.midi.receiver", "fluid.viewComponent"],
+
+    portType: "input",
+
+    preferredPort: undefined,
+
+    components: {
+        midiPortSelector: {
+            type: "flock.midi.portSelectorView",
+            container: "{that}.container",
+            options: {
+                portType: "{connectorView}.options.portType",
+                preferredPort: "{connectorView}.options.preferredPort",
+                events: {
+                    onPortSelected: "{connectorView}.events.onPortSelected"
+                }
+            }
+        },
+
+        connection: {
+            createOnEvent: "onValidPortSelected",
+            type: "flock.midi.connection",
+            options: {
+                openImmediately: true,
+                ports: {
+                    expander: {
+                        funcName: "flock.midi.connectorView.generatePortSpecification",
+                        args: [
+                            "{connectorView}.options.portType",
+                            "{midiPortSelector}.selectBox.model.selection"
+                        ]
+                    }
+                },
+
+                // TODO: These are ultimately midi.connection events.
+                // Is there a better way to distribute listeners from this
+                // parent "facade" object to its connection subcomponent?
+                events: {
+                    raw: "{connectorView}.events.raw",
+                    message: "{connectorView}.events.message",
+                    note: "{connectorView}.events.note",
+                    noteOn: "{connectorView}.events.noteOn",
+                    noteOff: "{connectorView}.events.noteOff",
+                    control: "{connectorView}.events.control",
+                    program: "{connectorView}.events.program",
+                    aftertouch: "{connectorView}.events.aftertouch",
+                    pitchbend: "{connectorView}.events.pitchbend"
+                },
+
+                listeners: {
+                    "onCreate.fireAfterConnectionOpen": {
+                        func: "{connectorView}.events.afterConnectionOpen.fire"
+                    }
+                }
+            }
+        }
+    },
+
+    events: {
+        onPortSelected: null,
+        onValidPortSelected: null,
+        afterConnectionOpen: null
+    },
+
+    listeners: {
+        "onPortSelected.validatePortSelection": {
+            funcName: "flock.midi.connectorView.validatePortSelection",
+            args: ["{that}"]
+        }
+    }
+});
+
+flock.midi.connectorView.generatePortSpecification = function (portType, portIDs) {
+    var spec = {};
+    spec[portType] = {
+        id: portIDs
+    };
+
+    return spec;
+};
+
+flock.midi.connectorView.validatePortSelection = function (that) {
+    var selectedId = fluid.get(that, "midiPortSelector.selectBox.model.selection");
```

Comment:
Seems odd to gear from this seemingly unrelated event "onPortSelected" back into a model interaction again

Reply:
@the-t-in-rtf This was from your original implementation, can you respond to this?

Reply:
It looks like it's a kind of gating function to ensure that however many times `onPortSelected` is fired, the component-creating even `validPortSelected` is only fired if there is a selected port available.  I'm having trouble remembering why it was needed.  Once I verify your recent changes working, I can bypass this check and see what happens.

Reply:
I suspect it was plastering over some misunderstanding about the inner workings of one or both of the preexisting components.  If I bypass that and create the connection directly when `onPortSelected` is fired, it works for all variations I've tried, including:

1. When no devices are connected and the page is loaded. ("None" is selected on startup.)
2. When the preferred device is already connected. (The preferred device is selected on startup.)
3. When the preferred device is activated after the page loads. (The preferred device is selected.)
4. When the preferred device is deactivated after the page loads. (The selection reverts to "none".)

In all cases there are no errors and the behaviour (in parentheses above) is as expected.  So it really seems like you can just axe this.